vm/actor/src/builtin/miner/mod.rs

```diff
@@ -382,43 +371,36 @@ impl Actor {
                 })?;
 
                 // If the PoSt was successful, the declared recoveries should be restored
-                st.remove_faults(rt.store(), &mut declared_recoveries)
+                st.remove_faults(rt.store(), &declared_recoveries)
                     .map_err(|e| {
                         ActorError::new(
                             ExitCode::ErrIllegalState,
                             format!("failed to remove recoveries from faults: {}", e),
                         )
                     })?;
 
-                st.remove_recoveries(&mut declared_recoveries)
-                    .map_err(|e| {
-                        ActorError::new(
-                            ExitCode::ErrIllegalState,
-                            format!("failed to remove recoveries: {}", e),
-                        )
-                    })?;
+                st.remove_recoveries(&declared_recoveries).map_err(|e| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalState,
+                        format!("failed to remove recoveries: {}", e),
+                    )
+                })?;
 
                 // Load info for recovered sectors for recovery of power outside this state transaction.
-                match declared_recoveries.is_empty() {
-                    Ok(true) => Ok(st.info.sector_size),
-                    Ok(false) => {
-                        let mut sectors_by_number: HashMap<SectorNumber, SectorOnChainInfo> =
-                            HashMap::new();
-                        for sec in sector_infos {
-                            sectors_by_number.insert(sec.info.sector_number, sec);
-                        }
-                        let _ = declared_recoveries.for_each(|i| {
-                            let key: SectorNumber = i as u64;
-                            let s = sectors_by_number.get(&key).cloned().unwrap();
-                            recovered_sectors.push(s);
-                            Ok(())
-                        });
-                        Ok(st.info.sector_size)
+                if declared_recoveries.is_empty() {
+                    Ok(st.info.sector_size)
+                } else {
+                    let mut sectors_by_number: HashMap<SectorNumber, SectorOnChainInfo> =
+                        HashMap::new();
+                    for sec in sector_infos {
+                        sectors_by_number.insert(sec.info.sector_number, sec);
                     }
-                    Err(e) => Err(ActorError::new(
-                        ExitCode::ErrIllegalState,
-                        format!("failed to check if bitfield was empty: {}", e),
-                    )),
+                    declared_recoveries.iter().for_each(|i| {
+                        let key = i as u64;
+                        let s = sectors_by_number.get(&key).cloned().unwrap();
+                        recovered_sectors.push(s);
```

Comment:
Sorry, unrelated to your changes, but can you add a TODO here to handle deduplication of values? 

I didn't notice before but it seems possible that there could be inconsistencies because the go impl puts in a hashmap before iterating and pushing to recovered_sectors. Feel free to make change here yourself if you want btw

Reply:
What would the keys of that hash map be? You already can't have two sectors for the same `key` in the code you commented on because they're pulled from a hash map, but I don't know what the Go code looks like so I might be missing the point

Reply:
oops, my bad. I misread skimming quickly and comparing with specs-actors. Ignore that no need to add the TODO